pkg/controller/deployment/sync.go

```diff
@@ -173,7 +173,7 @@ func (dc *DeploymentController) getNewReplicaSet(d *apps.Deployment, rsList, old
 
 		if needsUpdate {
 			var err error
-			if d, err = dc.client.AppsV1().Deployments(d.Namespace).UpdateStatus(context.TODO(), d, metav1.UpdateOptions{}); err != nil {
+			if _, err = dc.client.AppsV1().Deployments(d.Namespace).UpdateStatus(context.TODO(), d, metav1.UpdateOptions{}); err != nil {
```

Comment:
This `d` is being used later on in code, so it looks to me this is wrong change.

Reply:
From the code 

https://github.com/kubernetes/kubernetes/blob/da54185c9eb076a71962ddb5aded24811467dbb3/pkg/controller/deployment/sync.go#L149-L188

Here, the assignment of `d` will only be executed within the condition of `if existingNewRS != nil`, and then return directly to jump out of the function. The code segment used behind `d` is not within this logical scope, and the latter `d` is passed in by the current function

It seems like this is okay to modify

Reply:
Yeah, you're right, thx.